roles/matrix-prometheus-node-exporter/tasks/init.yml

```diff
@@ -23,10 +23,10 @@
               resolver 127.0.0.11 valid=5s;
               set $backend "matrix-prometheus-node-exporter:9100";
               proxy_pass http://$backend/metrics;
+            {% elif matrix_prometheus_postgres_exporter_container_http_host_bind_port != '' %}
+              proxy_pass http://{{ matrix_prometheus_postgres_exporter_container_http_host_bind_port }}/metrics;
```

Comment:
`matrix_prometheus_postgres_exporter_container_http_host_bind_port` (and other `_host_bind_port` variables we have) can contain different kinds of values. For example:

- `1234` - bare port number
- `127.0.0.1:1234` - loopback interface + port number
- `0.0.0.0:1234` - all IPv4 interfaces + port number

Your quick fix is likely to work only for the `127.0.0.1:1234` case and generate invalid configuration for the others.

----

This can be solved by adding a new variable, which is defined in terms of `matrix_prometheus_postgres_exporter_container_http_host_bind_port` somehow.

Example `roles/matrix-prometheus-postgres-exporter/vars/main.yml`:

```yaml
# `matrix_prometheus_postgres_exporter_container_http_host_bind_port_number_raw` contains the raw port number extracted from `matrix_prometheus_postgres_exporter_container_http_host_bind_port`,
# which can contain values like this: ('1234', '127.0.0.1:1234', '0.0.0.0:1234')
matrix_prometheus_postgres_exporter_container_http_host_bind_port_number_raw: "{{ '' if matrix_prometheus_postgres_exporter_container_http_host_bind_port == '' else (matrix_prometheus_postgres_exporter_container_http_host_bind_port.split(':')[1] if ':' in matrix_prometheus_postgres_exporter_container_http_host_bind_port else matrix_prometheus_postgres_exporter_container_http_host_bind_port) }}
```

Example `roles/matrix-prometheus-postgres-exporter/defaults/main.yml`:

```yaml
# proxy_pass host to use when matrix-nginx-proxy is NOT enabled.
matrix_prometheus_postgres_exporter_matrix_nginx_proxy_not_enabled_proxy_pass_host: "{{ '' if matrix_prometheus_postgres_exporter_container_http_host_bind_port_number_raw == '' else ('127.0.0.1' + matrix_prometheus_postgres_exporter_container_http_host_bind_port_number_raw) }}"
```

You can then use `matrix_prometheus_postgres_exporter_matrix_nginx_proxy_not_enabled_proxy_pass_host` in `tasks/init.yml`, after checking that it's not empty.